mtasa-resources/CODING_GUIDELINES.md

```diff
@@ -49,42 +59,92 @@ function exampleFunction(value)
 end
 ```
 
-### Consistent naming conventions
+### Error handling
 
-TODO
+Use the Lua [`error (message [, level])`](https://www.lua.org/manual/5.1/manual.html#pdf-error) function to raise warnings when validating parameters passed to a function, with `level` set to `2` so that the error points the error at the function call location. For example:
 
-### Use of constants
+```lua
+-- This function outputs player's name to the chat box
+function outputPlayerName(player)
+    -- If player is invalid, raise an error
+    if not (isElement(player) and getElementType(player) == "player") then
+        error("Invalid player!", 2)
+    end
+
+    outputChatBox(getPlayerName(player))
+end
 
-TODO
+-- This function is triggered each time a player joins and calls outputPlayerName
+local function playerJoin()
+    outputPlayerName() -- note the missing player argument; the error will point to this line
+end
+addEventHandler("onPlayerJoin", root, playerJoin)
+```
 
-### Indentation and formatting
+### Avoid anonymous functions
 
-Ensure your code editor (e.g. [Visual Studio Code](https://code.visualstudio.com/) 
-applies the rules established by the project's **.editorconfig** file.
+Rather than using anonymous functions (i.e when calling [`addEventHandler`](https://wiki.multitheftauto.com/wiki/AddEventHandler)), use a named local function instead. For example:
 
-# General principles
+```lua
+-- Rather than this:
+addEventHandler("onPlayerConnect", root, function()
+    outputChatBox("Hello!")
+end)
+
+-- Do this:
+local function onConnect()
+    outputChatBox("Hello!")
+end
+addEventHandler("onPlayerConnect", root, onConnect)
+```
```

Comment:
Anonymous functions are fine. One should not use anonymous functions if the function is used somewhere else too.

Reply:
I agree, we should phrase it more towards enforcement of not repeating logic, rather than avoiding anonymous functions.

Reply:
Dropped.